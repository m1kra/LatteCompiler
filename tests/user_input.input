Hello!
24
42
