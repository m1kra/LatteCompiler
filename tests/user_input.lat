
int main(){
    printString("Give me a string.");
    string s = readString();
    printString("The string I got is: " + s);
    printString("Give me two natural numbers.");
    int n = readInt(), m = readInt();
    printString("Their NWD is:");
    printInt(nwd(n, m));
    return 0;
}

int nwd(int a, int b){
    if(a > b){
        return nwd(a - b, b);
    } else {
        if(a < b)
            return nwd(a, b - a);
        return a;
    }
}