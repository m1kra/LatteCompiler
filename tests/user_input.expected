Give me a string.
The string I got is: Hello!
Give me two natural numbers.
Their NWD is:
6